test/e2e/far_e2e_test.go

```diff
@@ -253,17 +273,70 @@ func randomizeWorkerNode(nodes *corev1.NodeList) *corev1.Node {
 	return &nodes.Items[r.Intn(len(nodes.Items))]
 }
 
+// pickRemediatedNode returns a next remediated node from the current available nodes,
+// and then the node is removed from the list of available nodes
+func pickRemediatedNode(availableNodes *corev1.NodeList) (*corev1.Node, *corev1.NodeList) {
```

Comment:
wow, that's a lot of code for just selecting a worker node... but ok. One nit though: why returning the same node list which you get in the parameter? It's a pointer, you can modify the list as needed inside the function...

Reply:
Simplified the current logic and change it to modify the list inside the function.